src/main/java/com/apptopia/emr/javaagent/AddRequireToClinitClassVisitor.java

```diff
@@ -0,0 +1,69 @@
+package com.apptopia.emr.javaagent;
+
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+
+import java.net.URL;
+
+import static org.objectweb.asm.Opcodes.ACC_STATIC;
+import static org.objectweb.asm.Opcodes.RETURN;
+
+public final class AddRequireToClinitClassVisitor extends ClassVisitor {
+    private final ClassLoader loader;
+
+    private String className;
+    private int version;
+    private boolean skip;
+
+    private boolean visitedStaticBlock = false;
+
+    AddRequireToClinitClassVisitor(int api, ClassVisitor visitor, ClassLoader loader) {
+        super(api, visitor);
+        this.loader = loader;
+    }
+
+    @Override
+    public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+        super.visit(version, access, name, signature, superName, interfaces);
+        this.className = name;
+        this.version = version;
+        String scriptBase = className.split("\\$")[0];
+        skip = getResource(loader, scriptBase + "__init.class") == null
+                && getResource(loader, scriptBase + ".clj") == null
+                && getResource(loader, scriptBase + ".cljc") == null;
+//        if (!skip) {
```

Comment:
I think we can freely remove it or uncomment.

Reply:
How about checking an environment variable (e.g. `DEBUG_EMR_JAVAAGENT`) so the message is displayed only when we what it to?

Reply:
Sure. Any useful logging better than a commented block of code. 😃

Reply:
Done